### PR TITLE
use dateutil parser for datetime parsing

### PR DESCRIPTION
**Pull request by geobio**
the old regex wouldn't parse a datetime like: `2020-10-12 10:46:54.146488`
also, the regex is complicated and unnecessary

I replaced it to use `datetuil.parse()` which should handle everything. this was a suggestion from another pr, but not implemented because the suggestion was left after the pr was merged
https://github.com/opendoor-labs/rets/pull/41/files

### CODE REVIEW

File: rets/client/decoder.py [hunk 1]
```diff
@@ -68,14 +69,7 @@ def _get_decoder(data_type: str, interpretation: str, include_tz: bool = False):
...
 def _decode_datetime(value: str, include_tz: bool) -> datetime:
-    # Correct `0000-00-00 00:00:00` to `0000-00-00T00:00:00`
-    if re.match(r'^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}$', value):
-        value = '%sT%s' % (value[0:10], value[11:])
-    # Correct `0000-00-00` to `0000-00-00T00:00:00`
-    elif re.match(r'^\d{4}-\d{2}-\d{2}$', value):
-        value = '%sT00:00:00' % value[0:10]
-
-    decoded = udatetime.from_string(value)
+    decoded = dateutil.parser.parse(value)
```
Comment (blakeembrey) [inline rets/client/decoder.py:72]: Can we keep `udatetime` and RFC support? The `dateutil.parser.parse` library supports all sorts of random things and once we release this it'll be difficult to undo the change to make it stricter.

Comment (geobio) [reply rets/client/decoder.py]: i reverted to what you had before, and also added a test
now it does `if value[10] == ' ':`

**Review by blakeembrey — APPROVED** [review]
